ngraph/frontend/frontend_manager/include/frontend_manager/frontend_manager.hpp

```diff
@@ -0,0 +1,518 @@
+
+            /// \brief Convert only those parts of the model that can be converted leaving others
+            /// as-is. Converted parts are not normalized by additional transformations; normalize
+            /// function or another form of convert function should be called to finalize the
+            /// conversion process.
+            /// \param model Input model
+            /// \return partially converted nGraph function
+            virtual std::shared_ptr<ngraph::Function>
+                convert_partially(InputModel::Ptr model) const;
+
+            /// \brief Convert operations with one-to-one mapping with decoding nodes.
+            /// Each decoding node is an nGraph node representing a single FW operation node with
+            /// all attributes represented in FW-independent way.
+            /// \param model Input model
+            /// \return nGraph function after decoding
+            virtual std::shared_ptr<ngraph::Function> decode(InputModel::Ptr model) const;
+
+            /// \brief Runs normalization passes on function that was loaded with partial conversion
+            /// \param function partially converted nGraph function
+            virtual void normalize(std::shared_ptr<ngraph::Function> function) const;
+        };
+
+        /// Capabilities for requested FrontEnd
+        /// In general, frontend implementation may be divided into several libraries by capability
+        /// level It will allow faster load of frontend when only limited usage is expected by
+        /// client application as well as binary size can be minimized by removing not needed parts
+        /// from application's package
+        namespace FrontEndCapabilities
+        {
+            /// \brief Just reading and conversion, w/o any modifications; intended to be used in
+            /// Reader
+            static const int FEC_DEFAULT = 0;
+
+            /// \brief Topology cutting capability
+            static const int FEC_CUT = 1;
+
+            /// \brief Query entities by names, renaming and adding new names for operations and
+            /// tensors
+            static const int FEC_NAMES = 2;
+
+            /// \brief Partial model conversion and decoding capability
+            static const int FEC_WILDCARDS = 4;
+        }; // namespace FrontEndCapabilities
+
+        // -------------- FrontEndManager -----------------
+        using FrontEndCapFlags = int;
+        using FrontEndFactory = std::function<FrontEnd::Ptr(FrontEndCapFlags fec)>;
+
+        /// \brief Frontend management class, loads available frontend plugins on construction
+        /// Allows load of frontends for particular framework, register new and list available
+        /// frontends This is a main frontend entry point for client applications
+        class FRONTEND_API FrontEndManager
+        {
+        public:
+            FrontEndManager();
+
+            ~FrontEndManager();
```

Comment:
virtual destructor or final class?

Reply:
Set to final, no need for virtual table for this class